### PR TITLE
[Merged by Bors] - refactor: Use `Pairwise` wherever possible

### PR DESCRIPTION
**Pull request by eric-wieser**
Performed with a regex search for `∀ (.) (.), \1 ≠ \2 →`, and a few variants to catch implicit binders and explicit types.

I have deliberately avoided trying to make the analogous `Set.Pairwise` transformation  (or any `Pairwise (foo on bar)` transformations) in this PR, to keep the diff small.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
bors r-

bors d+

**Comment by jcommelin** [comment]
bors d=YaelDillies

File: Mathlib/Data/ZMod/Quotient.lean [hunk 1]
```diff
@@ -70,9 +70,9 @@ open BigOperators Ideal
 /-- The **Chinese remainder theorem**, elementary version for `ZMod`. See also
 `Mathlib.Data.ZMod.Basic` for versions involving only two numbers. -/
 def ZMod.prodEquivPi {ι : Type*} [Fintype ι] (a : ι → ℕ)
-    (coprime : ∀ i j, i ≠ j → Nat.Coprime (a i) (a j)) : ZMod (∏ i, a i) ≃+* ∀ i, ZMod (a i) :=
-  have : ∀ (i j : ι), i ≠ j → IsCoprime (span {(a i : ℤ)}) (span {(a j : ℤ)}) :=
-    fun i j h ↦ (isCoprime_span_singleton_iff _ _).mpr ((coprime i j h).cast (R := ℤ))
+    (coprime : Pairwise fun i j => Nat.Coprime (a i) (a j)) : ZMod (∏ i, a i) ≃+* ∀ i, ZMod (a i) :=
+  have : Pairwise fun i j => IsCoprime (span {(a i : ℤ)}) (span {(a j : ℤ)}) :=
+    fun i j h ↦ (isCoprime_span_singleton_iff _ _).mpr ((coprime h).cast (R := ℤ))
```
Comment (eric-wieser) [inline Mathlib/Data/ZMod/Quotient.lean:75]:
```suggestion
    fun _i _j h ↦ (isCoprime_span_singleton_iff _ _).mpr ((coprime h).cast (R := ℤ))
```

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**